src/main/webapp/script.js

```diff
@@ -38,3 +38,33 @@ function createCommentElement(comment) {
 function addParagraph(content) {
   return "<p>" + content + "</p>";
 }
+
+function newGroupchat() {
+  const request = new Request("/load-groupchat", {method: 'POST'});
+  fetch(request).then(response => response.text()).then((key) => {
+    redirectToGroupchat(key);
+  });
+}
+
+function loadGroupchat() {
+  const urlParams = new URLSearchParams(window.location.search);
+  const key = urlParams.get('key');
+  fetch('/load-groupchat?key=' + key).then(response => response.json())
+    .catch(error => {
+      alert('Error: Groupchat does not exist');
+      window.location.href = 'index.html';
+    }).then((messages) => {
+      const messageContainer = document.getElementById('messages');
+      messageContainer.innerHTML = '';
```

Comment:
Does this have a script injection concern :0

Reply:
I'm not entirely sure, but I don't think so because the message's innerText is set, and not its innerHTML. I'm not sure if this makes it safe of a script injection though